android/src/main/kotlin/com/aheaditec/freerasp/handlers/MethodCallHandler.kt

```diff
@@ -15,11 +20,31 @@ import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 internal class MethodCallHandler : MethodCallHandler {
     private var context: Context? = null
     private var methodChannel: MethodChannel? = null
+    private var pigeonApi: TalsecPigeonApi? = null
 
     companion object {
         private const val CHANNEL_NAME: String = "talsec.app/freerasp/methods"
     }
 
+    private val sink = object : MethodSink {
+        override fun onMalwareDetected(packageInfo: List<SuspiciousAppInfo>) {
+            context?.let { context ->
+                val pigeonPackageInfo = packageInfo.map { it.toPigeon(context) }
+                pigeonApi?.onMalwareDetected(pigeonPackageInfo) { result ->
+                    // Parse the result (which is Unit so we can ignore it) or throw an exception
+                    // Exceptions are translated to Flutter errors automatically
+                    result.getOrElseThenThrow {
+                        Log.e("MethodCallHandlerSink", "Result ended with failure")
+                    }
```

Comment:
```suggestion
                    result.getOrElse { ex ->
                        Log.e("MethodCallHandlerSink", "Result ended with failure")
                        throw ex
                    }
```

Imo, the extension method adds too much complexity. We could just keep it simple and inline it, what do you think?